scuba/config.py

```diff
@@ -261,9 +261,18 @@ def _get_volumes(data):
 
     vols = {}
     for cpath, v in voldata.items():
+        cpath = _expand_path(cpath)
         vols[cpath] = ScubaVolume.from_dict(cpath, v)
     return vols
 
+def _expand_path(in_str):
+    try:
+        output = expand_env_vars(in_str)
+    except (KeyError, ValueError):
```

Comment:
The [description](https://docs.python.org/3/library/string.html#string.Template.safe_substitute) for `string.Template.safe_substitute` includes the following:
>  Also, unlike with `substitute()`, any other appearances of the `$` will simply return `$` instead of raising `ValueError`.

The implementation of `string.Template.substitute` (taken here from the Python 3.10 source) also raises a ValueError exception in some cases:
```python3
def substitute(self, mapping=_sentinel_dict, /, **kws):
    if mapping is _sentinel_dict:
        mapping = kws
    elif kws:
        mapping = _ChainMap(kws, mapping)
    # Helper function for .sub()
    def convert(mo):
        # Check the most common path first.
        named = mo.group('named') or mo.group('braced')
        if named is not None:
            return str(mapping[named])
        if mo.group('escaped') is not None:
            return self.delimiter
        if mo.group('invalid') is not None:
            self._invalid(mo)
        raise ValueError('Unrecognized named group in pattern',
                         self.pattern)
    return self.pattern.sub(convert, self.template)
```

Therefore, I think we should catch both exceptions in `_expand_path`, though we may need to catch them individually to accommodate your request in https://github.com/JonathonReinhart/scuba/pull/192#discussion_r782752537 and provide good error messages for each one.

Reply:
As of 2405ed8 there is now a test that shows one condition where a `ValueError` can be raised from the `utils.expand_env_vars` function:
https://github.com/JonathonReinhart/scuba/blob/191-expand-vars-in-volumes/tests/test_utils.py#L147

Reply:
Nice catch! And crappy docs.

I confirmed that this code will raise `ValueError: Invalid placeholder in string: line 1, col 5`
```python
Template('foo $ bar').substitute()
```